src/cljs/harja/ui/checkbox.cljs

```diff
@@ -31,21 +31,23 @@ checkbox-tila->luokka {:valittu "harja-checkbox-valittu"
   ([tila-atom nimi] (checkbox tila-atom nimi {}))
   ([tila-atom nimi {:keys [on-change width otsikon-luokka] :as optiot}]
    (let [tila @tila-atom
-         vaihda-tila (fn []
+         vaihda-tila (fn [event]
                        (let [uusi-tila (case tila
                                          :valittu :ei-valittu
                                          :ei-valittu :valittu
                                          :osittain-valittu :ei-valittu)]
                          (reset! tila-atom uusi-tila)
                          (when on-change
-                           (on-change uusi-tila))))]
+                           (on-change uusi-tila))
+                         (.stopPropagation event)))]
      [:div.harja-checkbox
       [:div.harja-checkbox-sisalto {:style {:width (or width "100%")}
-                                    :on-click (fn [event]
-                                                (vaihda-tila)
-                                                (.stopPropagation event))}
+                                    :on-click #(vaihda-tila %)}
        [:div.harja-checkbox-column
-        [:div.harja-checkbox-laatikko {:class (checkbox-tila->luokka tila)}
+        [:div.harja-checkbox-laatikko {:class (checkbox-tila->luokka tila)
+                                       :tabIndex "0"
+                                       :on-key-down #(when (= 32 (-> % .-keyCode))
```

Comment:
Jep. Hyvä ehdotus Samulta. Utilityfunktio, joka on nimetty entteriksi tai tabiksi on kyllä helpompi lukea, kuin maaginen numero 32.

Reply:
Hei muuten, tuo 32 onkin välilyöntinäppäin ja se on toistaiseksi käytössä vain tuossa kohdassa kun checkbox pitää saada valittua näppäimistöllä päälle ja pois. Jos näitä löytyy muualtakin (näitä itse tehtyjä checkboxeja) niin laitan sitten tästäkin oman utilityfunktion. Nyt tein vain tuosta enterin painalluksesta oman funktion.